torch/_inductor/scheduler.py

```diff
@@ -405,6 +416,59 @@ def can_inplace(self, read_dep: dependencies.MemoryDep):
             return read_dep.index == write_dep.index and read_dep.size == write_dep.size
         return False
 
+    def reorder_loop(
+        self, reorder: Callable[[sympy.Expr], sympy.Expr], ndim: int
+    ) -> Optional["SchedulerNode"]:
+        """Returns a new node which represents the same loop iterated in a new order
+
+        reorder is a function that takes indices in the new order and
+        translates them into the indices expected by the underlying loop. e.g.
+
+            reorder = lambda index: index[1], index[0]
+
+        will swap the order of a 2d loop. Note this only applies to the
+        pointwise index variables.
+
+        Also note some class members are shallow copied to the new node, so do
+        not modify the returned node.
+
+        """
+
+        if len(self.get_ranges()[0]) != ndim:
+            return None
+
+        def loop_reorder(ir_node: ir.ComputedBuffer):
+            (
+                iter_vars,
+                reduce_vars,
+            ), var_ranges = dependencies.index_vars_no_squeeze(
+                *self._sizes, prefix="sched_reorder"
+            )
+            assert len(reduce_vars) == 0
+            body = ir.LoopBody(
+                self._body,
+                [reorder(iter_vars), reduce_vars],
```

Comment:
Usage of redorder doesn't seem to match what is in the docs above.

Reply:
Good catch, the bug didn't show up with the 2d cases I was testing.